sdk/src/main/scala/com/horizen/account/network/SyncStatusActor.scala

```diff
@@ -0,0 +1,227 @@
+      .asInstanceOf[Int]
+
+    // if highest timestamp from the batch of new blocks is greater than the timestamp from the node current block
+    // the new highest block value will be calculated and the sync status updated, otherwise we will ignore this new message
+    if (highestBlockTimestampFromMessage.compareTo(currentBlockFromView.timestamp) > 0) {
+
+      // calculate the timestamp difference between the highest block from the event and the current block
+      val timestampDifference = highestBlockTimestampFromMessage - currentBlockFromView.timestamp
+
+      // calculate, given the correction calculated on the past blocks, how many blocks separate the current block from the tip
+      val blockCorrection = Await.result(sidechainNodeViewHolderRef ? GetDataFromCurrentView(calculateBlockCorrection), timeoutDuration)
+        .asInstanceOf[Double]
+      val blocksFromCurrentToTip = ((timestampDifference / params.consensusSecondsInSlot.toDouble) * blockCorrection).toInt
+
+      // set the actor internal state
+      syncStatus = true
+      startingBlock = currentHeightFromView
+      highestBlock = currentHeightFromView + blocksFromCurrentToTip
+      highestBlockTimestamp = highestBlockTimestampFromMessage
+
+      // broadcast sync start and its details, starting block is equal to current block in this case
+      val syncStatusMessage = new SyncStatus(true, startingBlock, startingBlock, highestBlock)
+      context.system.eventStream.publish(NotifySyncStart(syncStatusMessage))
+
+    }
+  }
+
+  protected def processSidechainNodeViewHolderEvents: Receive = {
+
+    case GetHighestBlockTimestamp(blockTimestamp) =>
+      processNewHighestBlockTimestamp(blockTimestamp)
+
+    case SemanticallySuccessfulModifier(sidechainBlock: PMOD) =>
+      processNewBlockApplied(sidechainBlock)
+
+  }
+
+  // used at rpc level (eth_syncing method) to retrieve updated information on the sync status of the node
+  protected def returnSyncStatus: Receive = {
+    case ReturnSyncStatus() =>
+      if (syncStatus) {
+        // retrieve current block height from view
+        val currentHeightFromView = Await.result(sidechainNodeViewHolderRef ? GetDataFromCurrentView(retrieveCurrentHeight), timeoutDuration)
+          .asInstanceOf[Int]
+        sender() ! new SyncStatus(syncStatus, currentHeightFromView, startingBlock, highestBlock)
+      } else {
+        sender() ! new SyncStatus(syncStatus)
+      }
+  }
+
+  override def receive: Receive = {
+    processSidechainNodeViewHolderEvents orElse returnSyncStatus orElse {
+      case message: Any => log.error("SyncStatusActor received strange message: " + message.getClass.toString)
+    }
+  }
+
+}
+
+object SyncStatusActor {
+
+  object ReceivableMessages {
+
+    case class ReturnSyncStatus()
+
+    case class NotifySyncStart(syncStatus: SyncStatus)
+
+    case class NotifySyncStop(syncStatus: Boolean)
```

Comment:
why do we need any parameter here?
it is clear from the name the status - not syncing

```suggestion
    case object NotifySyncStop
```